codice/ddf
# DDF-4031 As a user I would like to be able to export a result set as a KML file so that I can have a kml file to use externally

#### What does this PR do?
The current KMLTransformerImpl (QueryResponseTransformer) exports attributes to the description tag. This also extracts attributes and injects them into extended data. 
#### Who is reviewing it? 

@Schachte 
@brianfelix 
@Bdthomson 
@jrnorth 
@bellcc
@willwill96 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@millerw8 
@bdeining
#### How should this be tested?

- Perform `/internal/cql/transform/kml` query with [DDF-3997](https://github.com/codice/ddf/pull/3516) and ensure KML file is in attachment header (web sockets off, populated with metacards).
- Validate KML data, including description and extended data.
#### Any background context you want to provide?
Relies on [DDF-3997](https://github.com/codice/ddf/pull/3516)
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
